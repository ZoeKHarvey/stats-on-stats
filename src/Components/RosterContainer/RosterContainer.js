import React, { Component } from 'react';
import TeamCard from '../TeamCard/TeamCard'
import { connect } from 'react-redux';
import RosterCard from '../RosterCard/RosterCard'
import './RosterContainer.scss'

export const RosterContainer = ({roster, getSinglePlayer}) => {
  console.log('roster in container-->', roster)
  const singleRoster = roster.map((roster) => {
    return <RosterCard {...roster}
                        getSinglePlayer = { getSinglePlayer } 
                        />
})

    return(
      <section className="section-rostercontainer">
        {singleRoster}
  
      </section>
    )
}

export const mapStateToProps = (state) => ({
  teams: state.teams,
  roster: state.roster
})

export default connect(mapStateToProps)(RosterContainer);