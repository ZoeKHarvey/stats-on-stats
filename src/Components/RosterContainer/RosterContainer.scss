.section-rostercontainer {
  background-image: url('../../images/ice5.jpg');
  background-size: cover;
  height: 100vh;
  display: flex;
  flex-wrap: wrap;
  overflow: scroll;
}

.roster__h3--away,
.roster__h3--home {
  font-weight: 800;
  font-size: 1.2em;
}

.roster__container--game {
  display: flex;
  flex-direction: column;
  width: 100%;
  text-align: center;
}

.roster__div--schedule {
  display: flex;
  justify-content: space-around;
  border: 1px solid black;
  padding-bottom: 2%;
}

.roster--h2--game {
  margin-left:auto;
  margin-right: auto;
}

.roster__h1--roster {
  border-bottom: 1px solid black;
  width: 100%;
  padding-bottom: 1.5%;
}