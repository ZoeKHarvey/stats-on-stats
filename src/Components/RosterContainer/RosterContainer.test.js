import React from 'react';
import { RosterContainer, mapStateToProps } from './RosterContainer';
import { shallow } from 'enzyme';

describe('RosterContainer', () => {
  let wrapper;

  let roster = [{
    jerseyNumber: 1,
    name: 'Veet',
  }];

  let teamSchedule = {
    home: 'Doggos',
    away: 'Cattans'
  };

  beforeEach(() => {
    wrapper = shallow(
      <RosterContainer
        roster={roster}
        teamSchedule={teamSchedule} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mockStateToProps', () => {
  it('should return an object with the roster data array', () => {
    const mockState = {
      roster: [{jerseyNumber: 1}, {jerseyNumber: 2}],
      teamSchedule: {home: 'here', away: 'there'}
    };
    const expected = {
      roster: [{jerseyNumber: 1}, {jerseyNumber: 2}],
      teamSchedule: {home: 'here', away: 'there'}
    };
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  })
})