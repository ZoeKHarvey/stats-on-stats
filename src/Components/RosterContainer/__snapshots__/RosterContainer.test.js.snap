// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RosterContainer should match the snapshot 1`] = `
<section
  className="section-rostercontainer"
>
  <div
    className="roster__container--game"
  >
    <h2
      className="roster--h2--game"
    >
      Today's Game
    </h2>
    <div
      className="roster__div--schedule"
    >
      <div
        className="roster__div--home"
      >
        <h3>
          Home Team
        </h3>
        <h3
          className="roster__h3--home"
        />
        <h4>
          Wins: undefined
        </h4>
        <h4>
          Losses: undefined
        </h4>
        <h4>
          OTL: undefined
        </h4>
      </div>
      <div
        className="roster__div--away"
      >
        <h3>
          Away Team
        </h3>
        <h3
          className="roster__h3--away"
        />
        <h4>
          Wins: undefined
        </h4>
        <h4>
          Losses: undefined
        </h4>
        <h4>
          OTL: undefined
        </h4>
      </div>
    </div>
    <h1
      className="roster__h1--roster"
    >
      Roster
    </h1>
  </div>
  <Connect(RosterCard)
    jerseyNumber={1}
    key="0"
    name="Veet"
    teamSchedule={
      Object {
        "away": "Cattans",
        "home": "Doggos",
      }
    }
  />
</section>
`;
